internal/pkg/gateway/commitstatus.go

```diff
@@ -33,6 +33,11 @@ func (gs *Server) CommitStatus(ctx context.Context, signedRequest *gp.SignedComm
 		return nil, status.Errorf(codes.InvalidArgument, "invalid status request: %v", err)
 	}
 
+	// Validate the request has valid channel id and transaction id
+	if request.GetIdentity() == nil || request.GetChannelId() == "" || request.GetTransactionId() == "" {
+		return nil, status.Error(codes.InvalidArgument, "request must have valid identity, channel ID and transaction ID")
+	}
+
```

Comment:
It would be better to have three separate checks to indicate the problem rather than letting users guess. Additionally, adding a unit test to ensure the fix works would benefit this PR.

Reply:
Will split, thanks!